### PR TITLE
Add missing usubscription type export, minor qol functions

### PR DESCRIPTION
**Pull request by AnotherDaniel**
Minor items:
- Missing export of UnsubscribeResponse type in usubscription module
- is_empty() checks for UUri and SusbcriberInfo types (will compare with ::default() objects)

### CODE REVIEW

**Review by sophokles73 — APPROVED** [review]

**Review by sophokles73 — COMMENTED** [review]
IMHO we will still need `extract_from_protobuf` functionality in UMessage because not all L2 API uses UPayload, e.g. the PubSub API uses UListener for handling messages ...

**Comment by AnotherDaniel** [comment]
@sophokles73 alright this is banged into shape now - can you have a final look?

**Comment by AnotherDaniel** [comment]
@sophokles73 alright this is banged into shape now - can you have a final look?

File: src/communication.rs [hunk 1]
```diff
@@ -114,13 +181,64 @@ impl UPayload {
...
+    /// type to `UPayload::UPAYLOAD_FORMAT_PROTOBUF`.
+    ///
+    /// # Arguments
+    ///
+    /// * `message` - The protobuf `Message` to wrap in UPayload.
+    ///
+    /// # Returns
+    ///
+    /// `UPayload` with `Message` as `UPayloadType::UPAYLOAD_FORMAT_PROTOBUF`` payload.
+    pub fn from_protobuf<M>(message: M) -> Option<Self>
```
Comment (sophokles73) [inline src/communication.rs:194]: IMHO this should better return a Result with an error if the message cannot be serialized. However, we might also consider implementing `TryFrom<M: Message>` instead ...

File: src/communication/rpc.rs [hunk 2]
```diff
@@ -91,6 +92,48 @@ pub trait RpcClient: Send + Sync {
     ) -> Result<Option<UPayload>, ServiceInvocationError>;
 }
 
+impl dyn RpcClient {
```
Comment (sophokles73) [inline src/communication/rpc.rs:95]: Shouldn't this simply be a default implementation in the trait instead?

File: src/umessage.rs [hunk 3]
```diff
@@ -64,10 +66,37 @@ impl From<&str> for UMessageError {
 }
 
 impl UMessage {
-    /// Extracts the payload-contained protobuf message from a `UMessage`.
+    // Deserialize a proto-generated `Message`-type from payload `Bytes`, according to `UPayloadFormat`
+    // Will only succeed if payload format is one of
+    // - `UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF`
+    // - `UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY`
+    pub(crate) fn deserialize_protobuf_bytes<T: Message + Default>(
```
Comment (sophokles73) [inline src/umessage.rs:73]: FMPOV we should move this out of the UMessage struct as it has meaning independently from it

**Review by sophokles73 — CHANGES_REQUESTED** [review]

File: src/communication/rpc.rs (hunk 2 above)
Comment (AnotherDaniel) [reply src/communication/rpc.rs:95]: Nope - those have to be statically resolvable at compile-time, which doesn't work with generic types.

File: src/communication.rs (hunk 1 above)
Comment (AnotherDaniel) [reply src/communication.rs]: Can't make the TryFrom work, conflict with generic implemenation from core... went for using Result<>.

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**